geomesa-features/geomesa-feature-kryo/src/main/scala/org/locationtech/geomesa/features/kryo/KryoFeatureSerializer.scala

```diff
@@ -15,6 +15,27 @@ import org.locationtech.geomesa.features.kryo.impl.LazyDeserialization.{Immutabl
 import org.locationtech.geomesa.features.kryo.impl.{KryoFeatureDeserialization, KryoFeatureSerialization}
 import org.opengis.feature.simple.SimpleFeatureType
 
+/**
+  * Kryo feature serialization and deserialization.
+  *
+  * The current serialization scheme (version 3):
+  *
+  * <ol>
+  *   <li>one byte containing the serialization version (3)</li>
+  *   <li>two bytes for a short containing the number of serialized attributes. this supports
+  *       appending attributes to the schema, as we know how many attributes were written when deserializing</li>
+  *   <li>a known number of bytes for metadata, which consists of:
+  *     <ol>
+  *       <li>2 byte short per attribute for the offset to the start of the attribute in the serialized bytes</li>
```

Comment:
This puts a roughly 65kb upper limit on the size of a serialized SimpleFeature, right?  v2 had a varInt, right?  (Making sure I'm understanding; no judgment here:))

Reply:
ah, i guess it would actually be 32kb since we're using signed shorts. for some reason in my head it was 32mb...  maybe this limit is too small